omnibus/files/private-chef-cookbooks/private-chef/recipes/config.rb

```diff
@@ -0,0 +1,82 @@
+# Copyright:: Copyright (c) 2015 Chef Software, Inc
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Capture old node attribute values (if there are any, that is) in
+# case we need them for comparison purposes for making changes or
+# updates
+#
+# TODO: extract this into something that add-ons can use; no sense
+# cargo-culting it around everywhere
+if File.exists?("/etc/opscode/chef-server-running.json")
+  old_config = JSON.parse(IO.read("/etc/opscode/chef-server-running.json"))
+
+  # We're stashing these outside the "private_chef" attributes tree to
+  # prevent us from carrying them along forever when we write out the
+  # chef-server-running.json file at the end of the run.
+  #
+  # For example, to access the old version of the attribute
+  # ['private_chef']['foo']['bar'], you'll look at
+  # ['previous_run']['foo']['bar'].
+  #
+  # Take care to check that ['previous_run'] exists, though, otherwise
+  # you'll run into trouble doing the initial Chef run.
+  #
+  # TODO: Provide an API for getting this information
+  node.consume_attributes({"previous_run" => old_config['private_chef']})
+end
+
+
+if File.exists?("/etc/opscode/chef-server.json")
+  Chef::Log.warn("Please move to /etc/opscode/chef-server.rb for configuration - /etc/opscode/chef-server.json is deprecated.")
```

Comment:
@marcparadise Actually, where do you see this being moved?  From what I can see, if you have this file, pretty much nothing will work right since we skip the rest of the configuration.  Perhaps we should exit! here and make the user fix it?